### ISSUE TITLE
exclude some source files in pyproject.toml

### ISSUE THREAD

**Issue by ThomasWaldmann** [borgbackup/borg#7828]
Found by @bket:
"""
Interestingly, several source files (*.c, *.h, *.pyx) are being packaged/installed (2.0.0b7 versus 2.0.0b6). I believe this is caused by https://github.com/borgbackup/borg/commit/61e96eb51f6c17452ff3a506e2187d20ca5d204f.
"""

Initially, I didn't find how to convert some metadata to pyproject.toml (and assumed that it would magically work as desired).

But now I found: https://setuptools.pypa.io/en/stable/userguide/datafiles.html

Rather new `setuptools` seems to support this in `pyproject.toml`:
```
[tool.setuptools.packages.find]
where = ["src"]

[tool.setuptools.exclude-package-data]
mypkg = [".gitattributes"]
```

**Comment by ThomasWaldmann** [comment]
Here's the corresponding stuff removed from setup.cfg in that changeset:
```
# ...except the source files which have been compiled (C extensions):
[options.exclude_package_data]
* =
    *.c
    *.h
    *.pyx

[options.packages.find]
where = src
```

**Comment by ThomasWaldmann** [comment]
@bket fixed! see #7832.
